File: 백준/Gold/28018. 시간이 겹칠까？/시간이 겹칠까？.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.IO;  


namespace Algorithm
{
    class Program
    {
        static void Main()
        {

            var sr = new StreamReader(Console.OpenStandardInput());
            var sw = new StreamWriter(Console.OpenStandardOutput()) { AutoFlush = false };

            int[] imos = new int[1_000_002];

            int N = int.Parse(sr.ReadLine());

            for (int i = 0; i < N; i++)
            {
                var input = sr.ReadLine().Split();
                int S = int.Parse(input[0]);
                int E = int.Parse(input[1]);

                imos[S] += 1;
                imos[E+1] -= 1;
            }

            for (int i = 1; i <= 1_000_000; i++)
            {
                imos[i + 1] += imos[i];
            }

            int Q = int.Parse(sr.ReadLine());
            var queries = sr.ReadLine().Split();

            foreach (var qs in queries)
            {
                int qq = int.Parse(qs);
                sw.WriteLine(imos[qq]);
            }





            sw.Flush();
            sw.Close();
            sr.Close();
        }
    }
}

File: 백준/Gold/28018. 시간이 겹칠까？/시간이 겹칠까？.py
import sys
input = sys.stdin.readline
write = sys.stdout.write

N = int(input())

imos = [0] * 1_000_002

for _ in range(N):
    S, E = map(int, input().split())

    imos[S] += 1
    imos[E+1] -= 1

for i in range(1, 1_000_001):
    imos[i+1] += imos[i]

_ = int(input())
queries = list(map(int, input().split()))
write('\n'.join(str(imos[q]) for q in queries))
